java/src/test/java/com/xxAMIDOxx/xxSTACKSxx/menu/api/v1/impl/DeleteCategoryControllerImplTest.java

```diff
@@ -0,0 +1,139 @@
+import com.xxAMIDOxx.xxSTACKSxx.menu.repository.MenuRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.xxAMIDOxx.xxSTACKSxx.menu.domain.CategoryHelper.createCategory;
+import static com.xxAMIDOxx.xxSTACKSxx.menu.domain.MenuHelper.createMenu;
+import static com.xxAMIDOxx.xxSTACKSxx.util.TestHelper.getBaseURL;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.http.HttpStatus.CONFLICT;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.OK;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@EnableAutoConfiguration(
+        exclude = {
+                CosmosDbRepositoriesAutoConfiguration.class,
+                CosmosAutoConfiguration.class
+        })
+@Tag("Integration")
+class DeleteCategoryControllerImplTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private TestRestTemplate testRestTemplate;
+
+    @MockBean
+    private MenuRepository repository;
+
+    @AfterEach
+    void tearDown() {
+        repository.deleteAll();
+    }
+
+    @Test
+    void testDeleteCategorySuccess() {
+        // Given
+        Menu menu = createMenu(1);
+        Category category = createCategory(0);
+        menu.setCategories(List.of(category));
+        when(repository.findById(eq(menu.getId()))).thenReturn(Optional.of(menu));
+
+        // When
+        var requestEntity = getObjectHttpEntity();
+
+        String requestUrl =
+                String.format("%s/v1/menu/%s/category/%s", getBaseURL(port), menu.getId(), category.getId());
+        var response =
+                this.testRestTemplate.exchange(requestUrl, HttpMethod.DELETE, requestEntity, ErrorResponse.class);
+
+        // Then
+        ArgumentCaptor<Menu> captor = ArgumentCaptor.forClass(Menu.class);
+        verify(repository, times(1)).save(menu);
+        then(response.getStatusCode()).isEqualTo(OK);
+    }
+
+    @Test
+    void testDeleteCategoryWithAnItem() {
```

Comment:
Don't think this would be a failure would it? Item(s) just get deleted.
Fair enough though. Makes a nice edge case

Reply:
according to the swagger spec if a category has an item we can't delete it